src/wifi.rs

```diff
@@ -543,7 +543,7 @@ impl<'d> WifiDriver<'d> {
     pub fn scan_n<const N: usize>(
         &mut self,
         scan_config: &config::ScanConfig,
-    ) -> Result<(heapless::Vec<AccessPointInfo, N>, usize), EspError> {
+    ) -> Result<heapless::Vec<AccessPointInfo, N>, EspError> {
```

Comment:
`usize` was there for a reason: to indicate to you that you've fetched potentially less number of  aps than found. I'm not sure we want to get rid of it.

Moreover, you've removed it from some signatures, but kept it in others. Why?

Reply:
> Moreover, you've removed it from some signatures, but kept it in others. Why?

I didn't removed it from the trait signatures because I agree with you to don't change the public API at this point.

I see the point with a different fetch size. But I think it's clear that the `N` is the max capacity and the `len()` the count of the found access-points.

Reply:
> usize was there for a reason: to indicate to you that you've fetched potentially less number of aps than found. I'm not sure we want to get rid of it.

Doesn't heapless essentially encode this information already? `len()` being the number of results actually found, `N` or `capacity()` being the max you could search for?

Reply:
> Doesn't heapless essentially encode this information already? len() being the number of results actually found, N or capacity() being the max you could search for?

It isn't restricted by `N` how many the esp searches. It searches always all (dynamically allocated on the heap intern). The `usize` gives how many are found. And the heapless `len()` how many are fetched. If `N` is larger than the amount of found aps `len()` is the same as `usize`. If `N` is smaller than the amount of found aps `usize` tells the user how many could be fetched.

I will test if it's possible to fetch the remaining aps if not all are fetched at first.